src/gebt_poc/solver.cpp

```diff
@@ -557,4 +557,67 @@ Kokkos::View<double**> CalculateStaticIterationMatrix(
     return iteration_matrix;
 }
 
+Kokkos::View<double*> ConstraintsResidualVector(
+    const Kokkos::View<double*> gen_coords, const Kokkos::View<double*> position_vector
+) {
+    auto rotation_0 = gen_alpha_solver::EulerParameterToRotationMatrix(
+        gen_alpha_solver::create_vector({gen_coords(3), gen_coords(4), gen_coords(5), gen_coords(6)})
```

Comment:
Avoid allocating a new view by creating a subview of `gen_coords` containing the quaternion and passing it to `EulerParameterToRotationMatrix`:

```c++
    // Create rotation matrix from gen coordinates quaternion
    auto q0 = Kokkos::subview(gen_coords, Kokkos::make_pair(3, 7));
    auto rotation_0 = gen_alpha_solver::EulerParameterToRotationMatrix(q0);
```